pkg/operator/controllers/dnsmasq/scripts/99-dnsmasq-restart.gotmpl

```diff
@@ -27,12 +27,29 @@ log_dns_files() {
     log "$(echo -n "/etc/resolv.conf.dnsmasq file metadata: ") $(ls -lZ /etc/resolv.conf.dnsmasq)"
 }
 
+sanity_check() {
+    if [[ ! -f /etc/resolv.conf ]]; then
+        log "/etc/resolv.conf is missing. Sanity check failed."
+        exit 1
+    elif [[ ! -f /etc/resolv.conf.dnsmasq ]]; then
+        log "/etc/resolv.conf.dnsmasq is missing. Sanity check failed."
+        exit 1
+    fi
+}
+
 if [[ $interface == eth* && $action == "up" ]] || [[ $interface == eth* && $action == "down" ]] || [[ $interface == enP* && $action == "up" ]] || [[ $interface == enP* && $action == "down" ]]; then
+    sanity_check
     log "$action happened on $interface, connection state is now $CONNECTIVITY_STATE"
     log "Pre dnsmasq restart file information"
     log_dns_files
     log "restarting dnsmasq now"
-    if systemctl try-restart dnsmasq --wait; then
+    
+    if systemctl is-active dnsmasq --quiet; then
```

Comment:
just for my understanding, we are expecting dnsmasq service to not be running due to the host servicing, so if its active then we exit out? should we try to kill the service and then restart it?

Reply:
This script is to a fix `/etc/resolv.conf` when it has been modified after `dnsmasq` is running.
IIRC this is to prevent a restart loop of dnsmasq, I'll update this comment once I find it.
@ventifus will also have background knowledge on why it's done this way.

Reply:
@Spencer-z During the node boot process eth0 will come up/down before `dnsmasq` does. This check is to prevent attempting to start `dnsmasq` too early in the boot process.

Reply:
thank you for the clarification